etcd-client/src/lib.rs

```diff
@@ -219,67 +225,153 @@ mod tests {
 
     #[test]
     fn test_all() -> Result<()> {
```

Comment:
Comment here to tell why we do the tests this way such as concurrent data access issue.